smartapps/juano2310/jawbone-up-connect.src/jawbone-up-connect.groovy

```diff
@@ -349,67 +353,39 @@ def uninstalled() {
 }
 
 def pollChild(childDevice) {
-    def member = state.member
-    generatePollingEvents (member, childDevice)
-}
+		def childMap = [ value: "$childDevice.device.deviceNetworkId}"]
 
-def generatePollingEvents (member, childDevice) {
-    // lets figure out if the member is currently "home" (At the place)
-    def urlgoals = "https://jawbone.com/nudge/api/users/@me/goals"
-    def urlmoves = "https://jawbone.com/nudge/api/users/@me/moves"
-    def urlsleeps = "https://jawbone.com/nudge/api/users/@me/sleeps"
-    def goals = null
-    def moves = null
- 	def sleeps = null
-    httpGet(uri: urlgoals, headers: ["Authorization": "Bearer ${state.JawboneAccessToken}" ]) {response ->
-        goals = response.data.data
-    }
-    httpGet(uri: urlmoves, headers: ["Authorization": "Bearer ${state.JawboneAccessToken}" ]) {response ->
-        moves = response.data.data.items[0]
-    }
+		def params = [
+				uri: 'https://jawbone.com',
+				path: '/nudge/api/users/@me/goals',
+				headers: ["Authorization": "Bearer ${state.JawboneAccessToken}" ],
+				contentType: 'application/json'
+		]
 
-    try { // we are going to just ignore any errors
-        log.debug "Member = ${member.first}"
-        log.debug "Moves Goal = ${goals.move_steps} Steps"
-        log.debug "Moves = ${moves.details.steps} Steps"
+		asynchttp_v1.get('responseGoals', params, childMap)
 
-        childDevice?.sendEvent(name:"steps", value: moves.details.steps)
-        childDevice?.sendEvent(name:"goal", value: goals.move_steps)
-    	//setColor(moves.details.steps,goals.move_steps,childDevice)
-    }
-    catch (e) {
-            // eat it
-    }
+		def params2 = [
+				uri: 'https://jawbone.com',
+				path: '/nudge/api/users/@me/moves',
+				headers: ["Authorization": "Bearer ${state.JawboneAccessToken}" ],
+				contentType: 'application/json'
+		]
+
+		asynchttp_v1.get('responseMoves', params2, childMap)
 }
 
-def generateInitialEvent (member, childDevice) {
-    // lets figure out if the member is currently "home" (At the place)
-    def urlgoals = "https://jawbone.com/nudge/api/users/@me/goals"
-    def urlmoves = "https://jawbone.com/nudge/api/users/@me/moves"
-    def urlsleeps = "https://jawbone.com/nudge/api/users/@me/sleeps"
-    def goals = null
-    def moves = null
- 	def sleeps = null
-    httpGet(uri: urlgoals, headers: ["Authorization": "Bearer ${state.JawboneAccessToken}" ]) {response ->
-        goals = response.data.data
-    }
-    httpGet(uri: urlmoves, headers: ["Authorization": "Bearer ${state.JawboneAccessToken}" ]) {response ->
-        moves = response.data.data.items[0]
-    }
+def responseGoals(response, dni) {
+		def childDevice = getChildDevice(dni.value)
+		def goals = response.json.data
```

Comment:
if the response is not successful (non 2xx) or has some other error, or if the json cannot be parsed from the response, a runtime exception would be thrown here. Not sure if you want to handle those cases and log them or try to recover somehow; if not and you are fine with a runtime exception here, then looks good.